File: flutter_myproject/lib/screens/garden/plamvarieties/palm_variety.dart
class PalmVariety {
  final String varietyId;
  final String varietyName;
  final String? scientificName;

  PalmVariety({
    required this.varietyId,
    required this.varietyName,
    this.scientificName,
  });

  factory PalmVariety.fromJson(Map<String, dynamic> json) => PalmVariety(
        varietyId: json['variety_id'] as String,
        varietyName: json['variety_name'] as String,
        scientificName: json['scientific_name'] as String?,
      );
}

File: my_project_server/models/palm_variety.js
const db = require('../libs/db_pool');

const palmVariety = {
  getAll: async () => {
    try {
      const query = `SELECT variety_id, variety_name, scientific_name FROM palm_variety ORDER BY variety_id`;
      const result = await db.query(query);
      const rows = Array.isArray(result[0]) ? result[0] : (result.data || result);
      return { isError: false, data: rows, errorMessage: "" };
    } catch (error) {
      return { isError: true, data: [], errorMessage: error.message };
    }
  }
};

module.exports = palmVariety;